app/src/main/java/com/incwelltechnology/lms/ui/auth/AuthViewModel.kt

```diff
@@ -0,0 +1,25 @@
+package com.incwelltechnology.lms.ui.auth
+
+import android.view.View
+import androidx.lifecycle.ViewModel
+import com.incwelltechnology.lms.data.repository.UserRepository
+
+class AuthViewModel : ViewModel() {
+    var username: String? = null
+    var password: String? = null
+
+    var authListener: AuthListener? = null
+
+    fun onLoginButtonClick(view: View) {
+        authListener!!.onStarted()
+        if (username.isNullOrEmpty() || password.isNullOrEmpty()) {
+            //failure
+            authListener?.onFailure("Username or Password is Empty")
+            return
+        }
+        //success
+        val loginResponse= UserRepository().userLogin(username!!,password!!)
+        authListener?.onSuccess(loginResponse)
```

Comment:
@iamsannyrai it will take time to send network request you are sending success just after making network call which will always success never fails.

Reply:
The term "success" mentioned there was just for the case where our fields cannot be empty or null. However for handling network requests and its state we have defined callback function in UserRepository class. These callback functions will handle network request. Actual network call is made in UserRepository class.